src/Stratis.Bitcoin.Features.PoA/FederationMember.cs

```diff
@@ -77,6 +77,39 @@ public CollateralFederationMember(PubKey pubKey, Money collateralAmount, string
         /// <summary>Mainchain address that should have the collateral.</summary>
         public string CollateralMainchainAddress { get; set; }
 
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            var item = obj as CollateralFederationMember;
+            if (item == null)
+                return false;
+
+            return this.PubKey.Equals(item.PubKey);
+        }
+
+        /// <inheritdoc />
+        public override int GetHashCode()
+        {
+            return this.PubKey.GetHashCode();
+        }
+
+        public static bool operator ==(CollateralFederationMember a, CollateralFederationMember b)
+        {
+            if (ReferenceEquals(a, b))
+                return true;
+
+            if (((object)a == null) || ((object)b == null))
+                return false;
+
+            return (a.PubKey == b.PubKey) && (a.CollateralAmount == b.CollateralAmount) && (a.CollateralMainchainAddress == b.CollateralMainchainAddress);
```

Comment:
`>=`?

Reply:
Or should this go in the caller?

Reply:
Why >= ? It's equality comparer so when amount is not the same then those are 2 different members. 

Btw it's not possible to have 2 members with identical pubkeys anyway so in all real life cases we will see those 2 values being the same